internal/deploy/bootstrap.go

```diff
@@ -0,0 +1,77 @@
+package deploy
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+
+	"github.com/kyma-project/cli/internal/kube"
+	"github.com/kyma-project/cli/internal/kustomize"
+)
+
+// Bootstrap deploys the kustomization files for the prerequisites for Kyma.
+// Returns true if the Kyma CRD was deployed.
+func Bootstrap(kustomizations []string, k8s kube.KymaKube, dryRun bool) (bool, error) {
+	defs := []kustomize.Definition{}
+	// defaults
+	if len(kustomizations) == 0 {
+		lm, err := kustomize.ParseKustomization("https://github.com/kyma-project/lifecycle-manager/operator/config/default")
```

Comment:
CRDs are included as part of the [default kustomization](https://github.com/kyma-project/lifecycle-manager/blob/01b4c7bd23b0f6df915b253b9890f6053f97582d/operator/config/default/kustomization.yaml#L23).

kustomize build includes them.

Reply:
This is not what I meant. I meant that kustomize by default does not order its resources. That means that the CR might be installed ahead of time from the CRD and error out in the first apply. This means you will fail out if you do not retry the apply. Here is an example run that fails because of this: https://github.com/kyma-project/lifecycle-manager/actions/runs/3202944355/jobs/5232482834

Reply:
I see, this is then a potential issue yes, currently all is applied together. Although the CLI k8s client parses the manifest from top to bottom and sends each resource to the async channel for it to be applied, there is no guarantee for the CRD to be there before the CR.

I will figure out how we can improve this.

Reply:
Exactly, I think basically you will need to apply retry logic, and / or apply a smarter ordering here.

Reply:
As discussed, I will solve the time matter with retry logic.